### PR TITLE
msft: use lisav2

### PR DESCRIPTION
**Pull request by ader1990**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -295,215 +364,132 @@ pipeline {
...
-                        -DBConfigPath "${env:DBConfigPath}"
-                        -LisaLogDBConfigPath "${env:DB_CONFIG_KERNEL}"
-                        -LogsPath "${env:LISA_KERNEL_LOG_DESTINATION}"
-                  ''')
-                echo 'Finished running LISA.'
-              }
+                println("Current VHD: ${env.HYPERV_VHD_PATH}")
+                RunPowershellCommand(".\\Run-LisaV2.ps1" +
+                    " -TestLocation 'localhost'" +
+                    " -RGIdentifier '${LISAV2_RG_IDENTIFIER}'" +
```
Comment (mbivolan) [inline Jenkinsfile]: is this groovy or ps? please use env. or env: so it's easily distinguishable

File: Jenkinsfile [hunk 2]
```diff
@@ -515,52 +501,35 @@ pipeline {
...
+                   variable: 'Azure_Secrets_File')
+            ]) {
+                prepareEnv(LISAV2_BRANCH, LISAV2_REMOTE, DISTRO_VERSION, FUNCTIONAL_TESTS)
+                unstash 'CapturedVHD.azure.env'
+                script {
+                    env.CapturedVHD = readFile 'CapturedVHD.azure.env'
+                }
+                println("VHD under test : ${env.CapturedVHD}")
+                RunPowershellCommand(".\\Run-LisaV2.ps1" +
+                    " -TestLocation '${LISAV2_AZURE_REGION}'" +
```
Comment (mbivolan) [inline Jenkinsfile]: ditto

File: Jenkinsfile [hunk 3]
```diff
@@ -295,215 +364,132 @@ pipeline {
...
-                        -DBConfigPath "${env:DBConfigPath}"
-                        -LisaLogDBConfigPath "${env:DB_CONFIG_KERNEL}"
-                        -LogsPath "${env:LISA_KERNEL_LOG_DESTINATION}"
-                  ''')
-                echo 'Finished running LISA for Debian Jessie.'
-              }
+                println("Current VHD: ${env.HYPERV_VHD_PATH}")
+                RunPowershellCommand(".\\Run-LisaV2.ps1" +
+                    " -TestLocation 'localhost'" +
+                    " -RGIdentifier '${LISAV2_RG_IDENTIFIER}'" +
```
Comment (mbivolan) [inline Jenkinsfile]: ditto

File: Jenkinsfile [hunk 4]
```diff
@@ -149,26 +185,24 @@ pipeline {
...
                         --install_deps "True" \\
-                        --thread_number "${THREAD_NUMBER}" \\
+                        --thread_number "x3" \\
+                        --debian_os_version "${UBUNTU_VERSION}" \\
                         --build_path "${BUILD_PATH}" \\
                         --kernel_config "${KERNEL_CONFIG}" \\
                         --clean_env "${CLEAN_ENV}" \\
-                        --use_ccache "${USE_CCACHE}"
+                        --use_ccache "${USE_CCACHE}" \\
+                        --enable_kernel_debug "${KERNEL_DEBUG}"
```
[the remark this answers is not in the document]
Comment (ader1990) [reply Jenkinsfile]: yes

File: Jenkinsfile [hunk 5]
```diff
@@ -32,28 +26,73 @@ def reportStageStatus(stageName, stageStatus) {
...
+def getVhdLocation(basePath, distroVersion) {
+    def distroFamily = distroVersion.split('_')[0]
+    return "${basePath}\\" + distroFamily + "\\" + distroVersion + "\\" + distroVersion + ".vhdx"
+}
+
+def prepareEnv(branch, remote, distroVersion, functionalTests) {
+    cleanWs()
+    git branch: branch, url: remote
+    script {
+      env.AZURE_OS_IMAGE = env.AZURE_UBUNTU_IMAGE_XENIAL
```
Comment (chvalean) [inline Jenkinsfile]: if it's Azure default to bionic then

File: Jenkinsfile [hunk 6]
```diff
@@ -32,28 +26,73 @@ def reportStageStatus(stageName, stageStatus) {
...
     choice(choices: 'False\nTrue', description: 'Enable kernel debug', name: 'KERNEL_DEBUG')
+    choice(choices: 'BVT\nFVT\nALL', description: 'Functional Tests', name: 'FUNCTIONAL_TESTS')
     string(defaultValue: "build_artifacts, publish_temp_artifacts, boot_test, publish_artifacts, publish_vhd, publish_azure_vhd, publish_hyperv_vhd, validation_functional_hyperv, validation_functional_jessie_hyperv, validation_functional_azure, validation_perf_azure, validation_perf_hyperv",
            description: 'What stages to run', name: 'ENABLED_STAGES')
   }
   environment {
+    LISAV2_REMOTE = "https://github.com/lis/LISAv2.git"
+    LISAV2_BRANCH = "master"
+    LISAV2_HYPERV_REGION = "hyper-v"
+    LISAV2_AZURE_REGION = "westeurope"
```
Comment (chvalean) [inline Jenkinsfile]: use westus2 by default

File: Jenkinsfile [hunk 7]
```diff
@@ -32,28 +26,73 @@ def reportStageStatus(stageName, stageStatus) {
...
+    LISAV2_RG_IDENTIFIER = "msftcbsl"
```
Comment (chvalean) [inline Jenkinsfile]: must change, msft_pipeline or something else

File: Jenkinsfile [hunk 8]
```diff
@@ -701,60 +563,39 @@ pipeline {
...
-                    build job: "performance-jobs/Pipeline_Performance", parameters: [
-                        string(name: 'KERNEL', value: "${kernelPath}")], wait: false;
-              }
-              echo "Triggered Local Hyper-V Performance tests."
-            
+            println("TBD")
+          }
+        }
+
+        stage('validation_sriov_hyperv') {
```
Comment (chvalean) [inline Jenkinsfile]: this needs to go after the validation stage, then followed by performance

Comment (ader1990) [reply Jenkinsfile]: will be done when impplemented

**Pull request merged**